### ISSUE TITLE
Ranger startup is delayed waiting for SMB mounts

### ISSUE THREAD

**Issue by jakem72360** [ranger/ranger#1899]
I have several SMB shares configured for automount under `/mnt` on my system. When not in use, systemd will unmount the shares, and remount them when needed. This is fine most of the time, however, I've noticed that launching Ranger, will always cause the shares to be remounted, even if I don't actually navigate to them. This is a frustrating UX issue, since Ranger will at best take >1s to start waiting for the shares to mount, and at worst, completely hang if the network is temporarily unavailable.

I'm wondering why Ranger would even try to touch filesystems that the user hasn't navigated to/opened files from. Does Ranger enumerate all mounts during launch, or is it because I have several directories on the shares bookmarked?

I'd be happy to help diagnose the issue myself (and maybe even draft a patch), but sadly I'm not very familar with Python's debugging facilities. If someone were willing to point me in the right direction, I'd be happy to try and offer some more help ;)

**Comment by zoqaeski** [comment]
This is an issue for me as well: I have a bunch of external drive automounts under `/mnt`, as well as using that location for extra internal drives that are normally mounted. Ranger will block until all the automounts in `/mnt` have mounted or timed out if I navigate to one of the internal drives one level down in the hierarchy, e.g. `/mnt/library`. Is there a way to prevent this from happening?

**Comment by toonn** [comment]
Like I said in the last comment, the only way is to hide those automounts a level deeper so ranger doesn't try `stat`ing them, as in one directory for each so `/mnt/xyz` becomes `/mnt/xyz/xyz`.

**Comment by jakem72360** [comment]
I just want to add that I don't mind Ranger stat'ing directory entries if I enter `/mnt` directly, after all, a `ls` on `/mnt` would do the same thing. I would also expect some delay when navigating to a mount while a connection is re-established. The issue here is that Ranger tries to stat directories that is isn't going to render in the current window. This also presents another problem; if the network is disconnected (or the connection dropped), Ranger will enter an uninterruptible sleep state when closing (via `:q`), or often when navigating. The terminal window can be forcefully closed, but the Python executable will remain, hung in state `D` until the connection comes back and the drives are remounted.

I don't see how changing filesystem structure would solve the issue since, as I said, Ranger hangs without navigating anywhere near the mountpoints. Even if it did work, it's a less than poor solution in my opinion, as it easily breaks symlinks, configurations, etc.

**Comment by toonn** [comment]
Hmm, I haven't experienced this. Are you starting ranger in `/`?

**Comment by jakem72360** [comment]
No, starting it in my home directory, but it still hangs waiting for mounts. Like I said in my original message, I'm not sure if it's because I have directories on the shares bookmarked? That's the only cause I could imagine.

**Comment by toonn** [comment]
That's probably it, yeah.
